### ISSUE TITLE
Wrong assigment is not flagged by compiler

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/TypeScript#19820]
**TypeScript Version:**  2.6.1
**Code**

```ts
interface A {
	x: number;
}

interface B {
	y: number;
}

function onEvent(callback: (param: A) => void): void {
}

onEvent((param: A & B) => {
	console.log('Hello');
});
```

**Expected behavior:**

- passing the arrow function to onEvent should be flagged as an error

**Actual behavior:**

- it is not flagged.

**Comment by ahejlsberg** [comment]
It's an error in `--strictFunctionTypes` mode. See #18654.

**Comment by dbaeumer** [comment]
Thanks. Closing then.
